elementary/clients/dbt/dbt_runner.py

```diff
@@ -241,3 +242,28 @@ def ls(self, select: Optional[str] = None) -> list:
 
     def source_freshness(self):
         self._run_command(command_args=["source", "freshness"])
+
+    def get_installed_packages(self):
+        dbt_packages_folder = os.path.join(os.getcwd(), 'dbt_packages')
+        try:
+            folder_names = [name for name in os.listdir(dbt_packages_folder) if os.path.isdir(os.path.join(dbt_packages_folder, name))]
+            return folder_names
+        except FileNotFoundError:
+            logger.info("'dbt_packages' folder not found.")
+            return False
+
+    def _run_deps_if_needed(self):
+        packages_yaml_file = os.path.abspath('packages.yml')
+        with open(packages_yaml_file, 'r') as file:
+            packages_data = yaml.safe_load(file)
+        package_names = [package_entry.get('package', '').split('/')[-1] for package_entry in packages_data['packages']]
+        package_names = [package for package in package_names if package != '']
+        installed_packages = self.get_installed_packages()
+        if not installed_packages:
```

Comment:
What do you think about:

```py
        if set(required_package_names) != set(installed_package_names):
            ...
```

Reply:
Sounds good, this looks much clean than the previous method.